cpp/oneapi/dal/algo/louvain/backend/cpu/vertex_partitioning_default_kernel.hpp

```diff
@@ -22,22 +22,427 @@
 #include "oneapi/dal/backend/memory.hpp"
 #include "oneapi/dal/backend/interop/common.hpp"
 #include "oneapi/dal/backend/dispatcher.hpp"
+#include "oneapi/dal/table/detail/table_builder.hpp"
+#include "oneapi/dal/algo/louvain/backend/cpu/louvain_data.hpp"
 
 namespace oneapi::dal::preview::louvain::backend {
 using namespace oneapi::dal::preview::detail;
 using namespace oneapi::dal::preview::backend;
+using namespace oneapi::dal::backend::primitives;
 
-template <typename Cpu, typename EdgeValue>
+template <typename IndexType>
+inline void singleton_partition(IndexType* labels, std::int64_t vertex_count) {
+    for (std::int64_t v = 0; v < vertex_count; v++) {
```

Comment:
better to use  approach applied by Polina. max_id = vertex_count - 1 and iterate over int32 values without cast in the loop.

Reply:
also use index_type template param  or it is a useless template))